caikit_nlp/modules/text_classification/sequence_classification.py

```diff
@@ -0,0 +1,245 @@
+
+        Args:
+            model_path: str
+                Path to the model to be loaded.
+
+        Returns:
+            SequenceClassification
+                Instance of this class built from the on disk model.
+        """
+        loader = ModuleLoader(model_path)
+        resource = loader.load_module("sequence_classifier")
+        device = SequenceClassification._get_device()
+        return cls(resource=resource, device=device)
+
+    @classmethod
+    def bootstrap(cls, base_model_path: str) -> "SequenceClassification":
+        """Bootstrap a HuggingFace transformer-based sequence classification model
+
+        Args:
+            base_model_path: str
+                Path to the model to be loaded.
+        """
+        # Note: Must provide path to tokenizer if model_name is a path
+        # for resource use
+        resource = HFAutoSequenceClassifier.bootstrap(
+            model_name=base_model_path, tokenizer_name=base_model_path
+        )
+        device = SequenceClassification._get_device()
+        return cls(
+            resource=resource,
+            device=device,
+        )
+
+    ################################## Private Functions #########################################
+
+    def _get_scores(self, text: Union[str, List[str]]):
+        """Run tokenizer and model to get scores on text(s)
+
+        Args:
+            text: Union[str, List[str]]
+                Input string(s) to be used
+
+        Returns:
+            scores_dict
+                Dict with key label, and values as the array of scores,
+                each corresponding to text(s)
+        """
+        # Apply tokenizer
+        tokenized_text = self.tokenizer(
+            text,
+            padding="max_length",
+            truncation=True,
+            return_tensors="pt",  # PyTorch
+        )
+        # NOTE: no simple/efficient way to detect whether truncation
+        # happened since padding also occurs, and this can be applied on
+        # a batch of strings.
+        if self.device == "cuda":
+            tokenized_text = tokenized_text.to(self.device)
+        with torch.no_grad():
+            logits = self.model(**tokenized_text).logits
+
+        if not self.model.config.id2label:
+            log.warning(
+                "<NLP31047577W>",
+                "No id2label provided in model config. Defaulting to numeric labels",
+            )
+
+        softmax = torch.nn.Softmax(dim=1)
+        raw_scores = softmax(logits)
+        if self.device == "cuda":
+            scores = raw_scores.cpu().numpy()
+        else:
+            scores = raw_scores.numpy()
+
+        scores_dict = {}
+        num_labels = self.model.num_labels
+        for label_idx in range(num_labels):
+            if self.model.config.id2label:
+                label = self.model.config.id2label[label_idx]
+            else:
+                label = label_idx
+            label_scores = scores[:, label_idx]
```

Comment:
The shape is expected to be `(num_texts, labels)` - this is getting all the scores for that label (across all texts). Will add a shape check

Reply:
don't we need an argmax to essentially fetch the highest score for each of the text and then the `scores_dict` will contain information like:
```json
{
    "text1": {
         "label": "foo",
         "score": 0.99999
    }
}
```

Reply:
Hm no since we're returning all scores for all labels essentially (this is text classification - no filtering/looking for particular labels), here all the scores for one label across all texts will be put in `label_scores`, and `scores_dict` will contain a mapping of `{label: [score, score, ..]}` where the length of the `[score, score, ..]` will be the number of texts

technically because this is built around transformers sequence classifier we assume the logits shape already https://huggingface.co/docs/transformers/main_classes/output#transformers.modeling_outputs.SequenceClassifierOutput.logits